### DISCUSSION TITLE
Expose a prepare method

### DISCUSSION THREAD

**Discussion by algorithmiker** [powersync-ja/sqlite_async.dart#8] [Ideas]
First of all, thanks for the excellent package!
I finally don't have to start every project by `db.execute("PRAGMA journal_mode=WAL;")`.

Anyways, I very often seem to use the following pattern with sqlite3:
```dart
class SomeService {
  PreparedStatement? _prepared;
  SomeData run(Database db) {
    prepared ??= db.prepare("select * from sqlite_master", persistent: true);
    return db.select(prepared);
  }
}
```

I wonder if this could be done via sqlite_async

**Comment by rkistner** [comment]
Is the main goal for that performance, or are there additional benefits?

I've found that in most cases, the overhead of cross-isolate communication is much more than the overhead of SQLite parsing a query, so it won't have a major benefit for the asynchronous calls. That said, there may be some cases with particularly large queries where it does make a difference.

Right now, for the cases where performance is critical, I'd recommend using `computeWithDatabase()`. With this, you're able to use the `prepare`  methods from the underlying `sqlite3` library directly.
